### DISCUSSION TITLE
How to implement right-click or long press events on a widget

### DISCUSSION THREAD

**Discussion by wanglifeng95** [beeware/toga#1970] [Q&A]
After reviewing relevant documents, I still couldn't find how to implement right-click or long press on a widget to pop up a menu.How can we achieve it in Toga？

**Comment by freakboy3742** [comment]
Toga doesn't currently have an API for pop-up menus. It should be possible to implement such a feature using direct access to platform native APIs; however, that code won't be cross-platform.

**Reply by wanglifeng95** [reply]
Icon buttons, right-click or long press widget to pop up menus, and scrollContainer's sliding display content are ineffective on some platforms. If these aspects are improved, Toga can cope with the development of complex interactive apps. Toga is more stable than Kivy. Hope Toga's widgets are more diverse and features are more complete, which is great.

**Reply by wanglifeng95** [reply]
Image clickability is also a desired feature

**Reply by freakboy3742** [reply]
Most of those issues are new features which would certainly be nice to add; I can't provide any predictions for when they'll be implemented - but this is also an open source project, so if you want these features, the best way to get them is to contribute a fix. Clickable images and icon buttons in particular have been a subject of extensive discussion in the past; search the forums and tickets for those discussions.

You've mentioned the ScrollContainer issue you're having previously; we're in the process of auditing and building tests for that widget  right now..

**Reply by wanglifeng95** [reply]
Toga is a very good open source project, and I also want to improve it, but I don't have the ability to do such an important thing now, so I can only rely on you
